pkg/acsengine/engine.go

```diff
@@ -984,6 +987,27 @@ func getDataDisks(a *api.AgentPoolProfile) string {
 	return buf.String()
 }
 
+func getEtcdDisk(orchestratorType api.OrchestratorType, m *api.MasterProfile) string {
+	if orchestratorType != api.Kubernetes {
+		return ""
+	}
+	var buf bytes.Buffer
+	buf.WriteString("\"dataDisks\": [\n")
+	etcdDisks := `            {
+              "createOption": "Empty",
+              "diskSizeGB": "%d",
+              "lun": %d,
+              "name": "[concat(variables('masterVMNamePrefix'), copyIndex(),'-etcddisk')]",
+              "vhd": {
+                "uri": "[concat(reference(concat('Microsoft.Storage/storageAccounts/',variables('masterStorageAccountName')),variables('apiVersionStorage')).primaryEndpoints.blob,'vhds/',    variables('masterVMNamePrefix'),copyIndex(),'-etcddisk.vhd')]"
+               }
+            }`
+
+	buf.WriteString(fmt.Sprintf(etcdDisks, 1, 0))
+	buf.WriteString("\n          ],")
+	return buf.String()
```

Comment:
Another consideration is do we want to enable encryption at rest by default for etcd?  There have been a few questions around this on the issues.  I'm not sure what this does performance wise.